woocommerce/woocommerce-android
# Refresh FCM tokens periodically

Closes: #11671 Closes: #11364


### Description
After the PR https://github.com/woocommerce/woocommerce-android/pull/6535, we updated the logic to register devices with our servers on these conditions only:

> A) User logged in
> B) The device token has been regenerated (e.g. due to security reasons)
> C) User opened the app and their device is not registered (e.g. during the log in backend was unresponsive/internet connection was not stable etc.)

This logic didn't account for the case where an FCM token expires, and then it gets deleted by our servers (fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Syvo%2Schfu%2Qabgvsvpngvbaf%2Sfreivprf%2SChfuAbgvsvpngvbaQvfcngpure.cuc%3Sn%3Qgehr%26e%3Q9400rq7o%26u%3Q335%23335-og), for this reason, Google [recommends](https://firebase.blog/posts/2023/04/managing-cloud-messaging-tokens/#updating-registration-tokens) updating tokens periodically, and this is what this PR implements.

### Testing information
1. Monitor logcat, then launch the app.
2. Confirm that the following is logged:
```
> Refreshing FCM token
> FCM token retrieved
> Sending FCM token to our remote services
```
3. Close the app.
4. Advance your device's time by 7 days.
5. Open the app, and confirm the logs are printed again.
